receiver/zookeeperreceiver/internal/metadata/generated_metrics_v2.go

```diff
@@ -0,0 +1,735 @@
+		mb.metrics.ZookeeperMaxFileDescriptors.CopyTo(metrics.AppendEmpty())
+	}
+	if mb.config.ZookeeperOpenFileDescriptors.Enabled && mb.metrics.ZookeeperOpenFileDescriptors.Gauge().DataPoints().Len() > 0 {
+		mb.metrics.ZookeeperOpenFileDescriptors.CopyTo(metrics.AppendEmpty())
+	}
+	if mb.config.ZookeeperOutstandingRequests.Enabled && mb.metrics.ZookeeperOutstandingRequests.Gauge().DataPoints().Len() > 0 {
+		mb.metrics.ZookeeperOutstandingRequests.CopyTo(metrics.AppendEmpty())
+	}
+	if mb.config.ZookeeperPacketsReceived.Enabled && mb.metrics.ZookeeperPacketsReceived.Sum().DataPoints().Len() > 0 {
+		mb.metrics.ZookeeperPacketsReceived.CopyTo(metrics.AppendEmpty())
+	}
+	if mb.config.ZookeeperPacketsSent.Enabled && mb.metrics.ZookeeperPacketsSent.Sum().DataPoints().Len() > 0 {
+		mb.metrics.ZookeeperPacketsSent.CopyTo(metrics.AppendEmpty())
+	}
+	if mb.config.ZookeeperPendingSyncs.Enabled && mb.metrics.ZookeeperPendingSyncs.Gauge().DataPoints().Len() > 0 {
+		mb.metrics.ZookeeperPendingSyncs.CopyTo(metrics.AppendEmpty())
+	}
+	if mb.config.ZookeeperSyncedFollowers.Enabled && mb.metrics.ZookeeperSyncedFollowers.Gauge().DataPoints().Len() > 0 {
+		mb.metrics.ZookeeperSyncedFollowers.CopyTo(metrics.AppendEmpty())
+	}
+	if mb.config.ZookeeperWatches.Enabled && mb.metrics.ZookeeperWatches.Gauge().DataPoints().Len() > 0 {
+		mb.metrics.ZookeeperWatches.CopyTo(metrics.AppendEmpty())
+	}
+	if mb.config.ZookeeperZnodes.Enabled && mb.metrics.ZookeeperZnodes.Gauge().DataPoints().Len() > 0 {
+		mb.metrics.ZookeeperZnodes.CopyTo(metrics.AppendEmpty())
+	}
+
+	// Reset metric data points collection.
+	mb.clearMetrics()
+}
+
+// zookeeperApproximateDateSizeMetric builds new zookeeper.approximate_date_size metric.
+func (mb *MetricsBuilder) zookeeperApproximateDateSizeMetric() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.SetName("zookeeper.approximate_date_size")
+	metric.SetDescription("Size of data in bytes that a ZooKeeper server has in its data tree.")
+	metric.SetUnit("By")
+	metric.SetDataType(pdata.MetricDataTypeGauge)
+	return metric
+}
+
+// zookeeperConnectionsAliveMetric builds new zookeeper.connections_alive metric.
+func (mb *MetricsBuilder) zookeeperConnectionsAliveMetric() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.SetName("zookeeper.connections_alive")
+	metric.SetDescription("Number of active clients connected to a ZooKeeper server.")
+	metric.SetUnit("1")
+	metric.SetDataType(pdata.MetricDataTypeGauge)
+	return metric
+}
+
+// zookeeperEphemeralNodesMetric builds new zookeeper.ephemeral_nodes metric.
+func (mb *MetricsBuilder) zookeeperEphemeralNodesMetric() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.SetName("zookeeper.ephemeral_nodes")
+	metric.SetDescription("Number of ephemeral nodes that a ZooKeeper server has in its data tree.")
+	metric.SetUnit("1")
+	metric.SetDataType(pdata.MetricDataTypeGauge)
+	return metric
+}
```

Comment:
To remove one allocation we should change this to init model:

```
func (mb *MetricsBuilder) initZookeeperEphemeralNodes(metric pdata.Metric) {
	metric.SetName("zookeeper.ephemeral_nodes")
	metric.SetDescription("Number of ephemeral nodes that a ZooKeeper server has in its data tree.")
	metric.SetUnit("1")
	metric.SetDataType(pdata.MetricDataTypeGauge)
}
```

Reply:
Makes sense. Addressed here https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6686